src/tools/rest-backend/root_application.cpp

```diff
@@ -92,14 +100,75 @@ void RootApp::version ()
 
 	if (request ().request_method () == "GET")
 	{
+		bool error = false;
 		cppcms::json::value data;
 
-		data["elektra"]["version"] = KDB_VERSION;
-		data["elektra"]["major"] = KDB_VERSION_MAJOR;
-		data["elektra"]["minor"] = KDB_VERSION_MINOR;
-		data["elektra"]["micro"] = KDB_VERSION_MICRO;
+		// api version
 		data["api"] = ELEKTRA_REST_API_VERSION;
 
+		// elektra version
+		try
+		{
+			kdb::KDB kdb;
+			kdb::KeySet ks;
+			kdb::Key k;
+			kdb.get (ks, "system/elektra/version/constants");
+
+			k = ks.lookup ("system/elektra/version/constants/KDB_VERSION");
+			if (!k)
+			{
+				error = true;
+			}
+			else
+			{
+				data["elektra"]["version"] = k.getString ();
+			}
+
+			k = ks.lookup ("system/elektra/version/constants/KDB_VERSION_MAJOR");
+			if (!k)
+			{
+				error = true;
+			}
+			else
+			{
+				data["elektra"]["major"] = k.getString ();
+			}
+
+			k = ks.lookup ("system/elektra/version/constants/KDB_VERSION_MINOR");
+			if (!k)
+			{
+				error = true;
+			}
+			else
+			{
+				data["elektra"]["minor"] = k.getString ();
+			}
+
+			k = ks.lookup ("system/elektra/version/constants/KDB_VERSION_MICRO");
+			if (!k)
+			{
+				error = true;
+			}
+			else
+			{
+				data["elektra"]["micro"] = k.getString ();
+			}
+		}
+		catch (kdb::KDBException & e)
+		{
+			error = true;
+		}
+
+		if (error)
+		{
+			// in case we could not retrieve the run-time version,
```

Comment:
Yes, there is a reason: within the lib ERROR is reserved for `ELEKTRA_ASSERT`, otherwise there is no error (API misuse is the only error and it should be only triggered via assertions). I clarified it in 1cc8af24be035bafa0bf623bc942e3efff8b4f9d

You are right, however, for your scenario ERROR would be useful. But it is trivial to write, see bottom of src/include/kdblogger.h, just add the macro definitions in one of your headers.